Packs/Netskope/Integrations/NetskopeEventCollector/NetskopeEventCollector.py

```diff
@@ -29,203 +35,249 @@ class Client(BaseClient):
-        # otherwise we save the list that include the old ids together with the new event ids.
-        new_last_run[f'{event_type}-ids'] = new_events_ids or last_run_ids
+def print_event_statistics_logs(events: list, event_type: str):
+    """
+    Helper function for debugging purposes.
+    This function is responsible to print statistics about pulled events, like the amount of pulled events and the first event and
+    last event times.
 
-    demisto.debug(f'Setting new last run - {new_last_run}')
-    return new_events, new_last_run
+    Args:
+        events (list): list of events.
+        event_type (str): the type of events given in the list.
+    """
+    demisto.debug(f'__[{event_type}]__ - Total events fetched this round: {len(events)}')
+    if events:
+        event_times = f'__[{event_type}]__ - First event: {events[0].get("timestamp")} __[{event_type}]__ - Last event: ' \
+                      f'{events[-1].get("timestamp")}'
+        demisto.debug(event_times)
 
 
-''' COMMAND FUNCTIONS '''
+def is_execution_time_exceeded(start_time: datetime) -> bool:
+    """
+    Checks if the execution time so far exceeded the timeout limit.
 
+    Args:
+        start_time (datetime): the time when the execution started.
 
-def test_module(client: Client, api_version: str, last_run: dict, max_fetch: int) -> str:
+    Returns:
+        bool: true, if execution passed timeout settings, false otherwise.
+    """
+    end_time = datetime.utcnow()
+    secs_from_beginning = (end_time - start_time).seconds
+    demisto.debug(f'Execution length so far is {secs_from_beginning} secs')
 
-    fetch_events_command(client, api_version, last_run, max_fetch=max_fetch, is_command=True)
-    return 'ok'
+    return secs_from_beginning > EXECUTION_TIMEOUT_SECONDS
 
 
-def get_all_events(client: Client, last_run: dict, limit: int, api_version: str, is_command: bool) -> Tuple[list, dict]:
+def setup_last_run(last_run_dict: dict) -> dict:
     """
-    This Function is doing a pagination to get all events within the given start and end time.
-    Maximum events to get per a fetch call is 50,000 (MAX_SKIP)
+    Setting the last_tun object with the right operation to be used throughout the integration run.
+
     Args:
-        client: Netskope Client
-        last_run (dict): the last run
-        limit (int): the number of events to return
-        api_version (str): The API version: v1 or v2
-        is_command (bool): Are we running the commands or test_module or not
+        last_run_dict (dict): The dictionary of the last run to be configured
 
-    Returns (list): list of all the events from a start time.
+    Returns:
+        dict: the modified last run dictionary with the needed operation
     """
-    events_result = []
-    new_last_run = {}
-    if limit is None:
-        limit = MAX_EVENTS_PAGE_SIZE
+    first_fetch = int(arg_to_datetime('now').timestamp())  # type: ignore[union-attr]
```

Comment:
So the first_fetch will always be from now?
We won't support going few days back?

Reply:
Nope, per new standards for collectors design we don't need to support fetching backwards unless specifically asked